app/Http/Controllers/SalarySlipController.php

```diff
@@ -0,0 +1,288 @@
+            foreach ($approvedPeriods as $approvedPeriod) {
+                foreach ($approvedPeriod as $approvedDates) {
+                    $approvedLeaveDate[] = $approvedDates->format('Y-m-d');
+                }
+            }
+
+            $Approved = [];
+            foreach ($approvedLeaveDate as $leaveDate) {
+                if (date('m', strtotime($leaveDate)) == $currentMonth && in_array(Carbon::parse($leaveDate)->format('l'), json_decode($weekend->weekend)) == false) {
+                    $Approved[] = $leaveDate;
+                }
+            }
+            $employeeApprovedLeaves[$employee->id] = count($Approved);
+
+            $numberOfDays = cal_days_in_month(CAL_GREGORIAN, $currentMonth, Carbon::parse($id)->year);
+            $workingDays = 0;
+            $mothDays = 0;
+            for ($i = 1; $i <= $numberOfDays; $i++) {
+                $date = Carbon::parse($i.'-'.$currentMonth.'-'.Carbon::parse($id)->year)->toDateString();
+                $mothDays += 1;
+                if (in_array(Carbon::parse($date)->format('l'), json_decode($weekend->weekend)) == false) {
+                    $workingDays += 1;
+                }
+            }
+
+            /* Absents */
+            $absent = [];
+            for ($i = 1; $i <= $mothDays; $i++) {
+                $date = Carbon::parse($i.'-'.$currentMonth.'-'.Carbon::parse($id)->format('Y'))->toDateString();
+                if (! in_array($date, $presentDate) && in_array(Carbon::parse($date)->format('l'), json_decode($weekend->weekend)) == false && in_array(Carbon::parse($date)->toDateString(), $Approved) == false && in_array(Carbon::parse($date)->toDateString(), $unApproved) == false) {
+                    $absent[] = '';
+                }
+            }
+            $AbsentCount[$employee->id] = count($absent);
+            $approvedCount[$employee->id] = $employeeApprovedLeaves[$employee->id];
+            $unApprovedCount[$employee->id] = $employeeUnApprovedLeaves[$employee->id];
+
+            if ($employee->emloyment_status != 'probation') {
+                if ($approvedCount[$employee->id] > 1) {
+                    $approvedDeduction = (($employee->basic_salary / $workingDays) * (($approvedCount[$employee->id]) - 1));
+                } else {
+                    $approvedDeduction = 0;
+                }
+            } elseif ($employee->emloyment_status == 'probation') {
+                $approvedDeduction = $approvedCount[$employee->id];
+            }
+            $absentDeduction[$employee->id] = ($employee->basic_salary / $workingDays) * $AbsentCount[$employee->id];
+            $netPayables[$employee->id] = round(($employee->basic_salary - ($employeeUnApprovedLeaves[$employee->id] * 2) - $approvedDeduction - $absentDeduction[$employee->id]) + ($employee->bonus));
+
+            // $unApprovedCount[$employee->id]=$employeeUnApprovedLeaves[$employee->id];
+            // $approvedCount[$employee->id]=$employeeApprovedLeaves[$employee->id];
```

Comment:
@JoharAlam 
Please remove commented code.

Reply:
Ok will do it